packages/protocol/test/governance/lockedgold.ts

```diff
@@ -412,4 +421,159 @@ contract('LockedGold', (accounts: string[]) => {
+          [NULL_ADDRESS],
+          [0],
+          { from: accounts[2] }
+        )
+      })
+
+      it("should reduce account's locked gold balance", async () => {
+        assertEqualBN(await lockedGold.getAccountNonvotingLockedGold(account), value - penalty)
+        assertEqualBN(await lockedGold.getAccountTotalLockedGold(account), value - penalty)
+      })
+
+      it("should increase the reporter's locked gold", async () => {
+        assertEqualBN(await lockedGold.getAccountNonvotingLockedGold(reporter), reward)
+        assertEqualBN(await lockedGold.getAccountTotalLockedGold(reporter), reward)
+      })
+
+      it("should increase the community fund's gold", async () => {
+        assert.equal(await web3.eth.getBalance(mockGovernance.address), penalty - reward)
+      })
+    })
+
+    describe('when the account is removed from `isSlasher`', () => {
+      const penalty = value
+      const reward = value / 2
+      beforeEach(async () => {
+        await lockedGold.removeSlasher(accounts[2])
+      })
+
+      it('should revert', async () => {
+        await assertRevert(
+          lockedGold.slash(account, penalty, reporter, reward, [NULL_ADDRESS], [NULL_ADDRESS], [0])
+        )
+      })
+    })
+
+    describe('when the account has half voting and half nonvoting gold', () => {
+      const voting = value / 2
+      const nonVoting = value - voting
+      beforeEach(async () => {
+        await election.vote(group, voting, NULL_ADDRESS, NULL_ADDRESS)
+      })
+
+      describe('when the account is slashed for only its nonvoting balance', () => {
+        const penalty = nonVoting
+        const reward = penalty / 2
+        beforeEach(async () => {
+          await lockedGold.slash(
+            account,
+            penalty,
+            reporter,
+            reward,
+            [NULL_ADDRESS],
+            [NULL_ADDRESS],
+            [0],
+            { from: accounts[2] }
+          )
+        })
+
+        it("should reduce account's nonvoting locked gold balance", async () => {
+          assertEqualBN(
+            await lockedGold.getAccountNonvotingLockedGold(account),
+            nonVoting - penalty
+          )
+        })
+
+        it('should leave the voting locked gold', async () => {
+          assertEqualBN(await lockedGold.getAccountTotalLockedGold(account), value - penalty)
+          assertEqualBN(await election.getTotalVotesByAccount(account), voting)
+        })
+
+        it("should increase the reporter's locked gold", async () => {
+          assertEqualBN(await lockedGold.getAccountNonvotingLockedGold(reporter), reward)
+          assertEqualBN(await lockedGold.getAccountTotalLockedGold(reporter), reward)
+        })
+
+        it("should increase the community fund's gold", async () => {
+          assert.equal(await web3.eth.getBalance(mockGovernance.address), penalty - reward)
+        })
+      })
+
+      describe('when the account is slashed for its whole balance', () => {
```

Comment:
do we tests for the case where we are slashed by the `min(balance, penalty)`. we should add those

Reply:
done